bagitobjecttransfer/recordtransfer/tests/unit/test_utils.py

```diff
@@ -109,123 +127,326 @@ class FileCountingUtilityTests(TestCase):
+            with self.subTest():
+                result = accept_file(filename, filesize)
+                self.assertTrue(result["accepted"])
+
+    def test_accept_file_invalid_extension(self) -> None:
+        """Test that files with invalid extensions are rejected."""
+        param_list = [
+            "p",
+            "mp3",
+            "docxx",
+        ]
+        for extension in param_list:
+            with self.subTest():
+                result = accept_file(f"My File.{extension}", "9012")
+                self.assertFalse(result["accepted"])
+                self.assertIn("extension", result["error"])
+
+    def test_accept_file_missing_extension(self) -> None:
+        """Test that files without an extension are rejected."""
+        result = accept_file("My File", "209")
+        self.assertFalse(result["accepted"])
+        self.assertIn("extension", result["error"])
+
+    def test_invalid_size(self) -> None:
+        """Test that files with invalid sizes are rejected."""
+        param_list = [
+            "-1",
+            "-1000",
+            "One thousand",
+        ]
+        for size in param_list:
+            with self.subTest():
+                result = accept_file("My File.pdf", size)
+                self.assertFalse(result["accepted"])
+                self.assertIn("size is invalid", result["error"])
+
+    def test_empty_file(self) -> None:
+        """Test that empty files are rejected."""
+        result = accept_file("My File.pdf", "0")
+        self.assertFalse(result["accepted"])
+        self.assertIn("empty", result["error"])
+
+    def test_file_too_large(self) -> None:
+        """Test that files that are too large are rejected."""
+        # Max size is patched to 1 MiB
+        param_list = [
+            (1024**2) + 1,  # 1 MiB plus one byte
+            (1024**2) * 8,  # 8 MiB
+            (1024**2) * 32,  # 32 MiB
+            (1024**3),  # 1 GiB
+            (1024**4),  # 1 TiB
+        ]
+        for size in param_list:
+            with self.subTest():
+                result = accept_file("My File.pdf", size)
+                self.assertFalse(result["accepted"])
+                self.assertIn("File is too big", result["error"])
+
+    def test_file_exactly_max_size(self) -> None:
+        """Test that files that are exactly the max size are accepted."""
+        # Max size is patched to 1 MiB
+        result = accept_file("My File.pdf", ((1024**2) * 1))  # 1 MiB
+        self.assertTrue(result["accepted"])
+
+
+class TestAcceptSession(TestCase):
+    """Tests for accept_session method."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        """Set up class-level test data."""
+        super().setUpClass()
+        settings.ACCEPTED_FILE_FORMATS = {"Document": ["docx", "pdf"], "Spreadsheet": ["xlsx"]}
+        settings.MAX_TOTAL_UPLOAD_SIZE = 3  # MiB
+        settings.MAX_SINGLE_UPLOAD_SIZE = 1  # MiB
+        settings.MAX_TOTAL_UPLOAD_COUNT = 4  # Number of files
+
```

Comment:
I recommend patching these settings here instead:

```suggestion

@patch(
    "django.conf.settings.ACCEPTED_FILE_FORMATS",
    {"Document": ["docx", "pdf"], "Spreadsheet": ["xlsx"]},
)
@patch("django.conf.settings.MAX_TOTAL_UPLOAD_SIZE", 3)  # MiB
@patch("django.conf.settings.MAX_SINGLE_UPLOAD_SIZE", 1)  # MiB
@patch("django.conf.settings.MAX_TOTAL_UPLOAD_COUNT", 4)  # Number of files
class TestAcceptSession(TestCase):
    """Tests for accept_session method."""

```

Reply:
Addressed by:
https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/292/commits/7022bf6a87fa86a5e97c3b59f6fd272b4f96a937